### PR TITLE
ci: fix 2.7 PHPSTAN

### PR DESCRIPTION
**Pull request by vasilvestre**
Follow up #https://github.com/api-platform/core/pull/5905

### CODE REVIEW

File: src/Symfony/Routing/IriConverter.php [hunk 1]
```diff
@@ -112,7 +112,7 @@ public function getIriFromResource($item, int $referenceType = UrlGeneratorInter
             return null;
         }
 
-        if ($this->resourceMetadataCollectionFactory && isset($context['item_uri_template'])) {
+        if (isset($context['item_uri_template'])) {
```
Comment (vasilvestre) [inline src/Symfony/Routing/IriConverter.php:115]: The class always exists

File: src/Hydra/Serializer/CollectionNormalizer.php [hunk 2]
```diff
@@ -49,12 +48,11 @@ final class CollectionNormalizer implements NormalizerInterface, NormalizerAware
     private $contextBuilder;
     private $resourceClassResolver;
     private $iriConverter;
-    private $resourceMetadataCollectionFactory;
     private $defaultContext = [
         self::IRI_ONLY => false,
     ];
 
-    public function __construct(ContextBuilderInterface $contextBuilder, ResourceClassResolverInterface $resourceClassResolver, $iriConverter, ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null, array $defaultContext = [])
```
Comment (soyuka) [inline src/Hydra/Serializer/CollectionNormalizer.php:57]: bc we can't do this

File: src/Symfony/Bundle/Resources/config/hydra.xml [hunk 3]
```diff
@@ -56,7 +56,6 @@
             <argument type="service" id="api_platform.jsonld.context_builder" />
             <argument type="service" id="api_platform.resource_class_resolver" />
             <argument type="service" id="api_platform.iri_converter" />
-            <argument type="service" id="api_platform.metadata.resource.metadata_collection_factory" on-invalid="null" />
```
Comment (soyuka) [inline src/Symfony/Bundle/Resources/config/hydra.xml:59]: we can't remove an argument

File: src/Hydra/Serializer/CollectionNormalizer.php (hunk 2 above)
Comment (vasilvestre) [reply src/Hydra/Serializer/CollectionNormalizer.php:57]: Should I ignore this ?

File: src/Symfony/Routing/IriConverter.php [hunk 4]
```diff
@@ -112,7 +112,7 @@ public function getIriFromResource($item, int $referenceType = UrlGeneratorInter
             return null;
         }
 
-        if ($this->resourceMetadataCollectionFactory && isset($context['item_uri_template'])) {
```
Comment (soyuka) [inline src/Symfony/Routing/IriConverter.php:115]: revert

**Pull request merged**